RFCs/0051-exclude-operator.adoc

```diff
@@ -0,0 +1,1703 @@
+-- To filter out `field~i~`, we would have to list out all fields other than `field~i~`.
+SELECT
+    field~1~, ..., field~i-1~, field~i+1~, ..., field~n~ -- omit `field~i~` from tbl
+FROM
+    tbl
+----
+
+== Guide-level explanation
+
+=== BNF Grammar
+
+[source,ebnf]
+----
+<sfw query> ::=
+    <select clause>
+    <exclude clause>?  (* NEW *)
+    <from clause>
+    (* other clauses from <sfw query> in PartiQL spec figure 3 *)
+
+<fws query> ::=
+    <from clause>
+    (* other clauses same as fws query *)
+    <exclude clause>?  (* NEW *)
+    <select clause>
+    
+<exclude clause> ::=
+    EXCLUDE <exclude path> [, <exclude path>]...
+
+<exclude path> ::=
+    <identifier>
+  | <exclude path> <dot> <identifier>
+  | <exclude path> <left bracket> <int literal> <right bracket>
+  | <exclude path> <left bracket> <string literal> <right bracket>
+  | <exclude path> <dot> <asterisk>
+  | <exclude path> <left bracket> <asterisk> <right bracket>
+
+<identifier> ::= (* See identifier in PartiQL spec figure 4 *)
+
+<asterisk> ::= "*"
+<dot> :: = "."
+<left bracket> ::= "["
+<right bracket> ::= "]"
+----
+
+NOTE: Despite their similar syntax and naming, ``<exclude path>``s are different than PartiQL path expressions.
+
+=== Terminology
+* For an `<exclude path>`, we refer to the leftmost identifier as the 'root' and the other exclude path components as 'steps'.
++
+[source]
+----
+e.g. tableFoo.a[1].*[*].b['c']
+    |  root  |     steps     |
+----
++
+* We refer to the exclude steps as follows
+** `.<identifier>` - tuple attribute exclude step
+*** `.<quoted identifier>`/`[<string literal>]` - case-_sensitive_ tuple attribute
+*** `.<unquoted identifier>` - case-_insensitive_ tuple attribute
+** `[<int literal>]` - collection index exclude step
+** `.*` - tuple wildcard exclude step
+** `[*]` - collection wildcard exclude step
+
+=== Out of scope / assumptions
+
+* We restrict tuple attribute exclude steps to use string literals and collection index exclude steps to use int literals. Thus `<exclude paths>` are statically known. We can decide whether to add other exclude paths (e.g. expressions) if a use case arises.
+* If sufficient schema is present and the path can be resolved, we assume the root of an `EXCLUDE` path can be omitted. The variable resolution rules follow what is already included in the PartiQL specification.
+* We require that every fully-qualified `<exclude path>` contain a root and at least one step. If a use case arises to exclude a binding tuple variable, then this functionality can be added.
+* S-expressions are part of the Ion type system.footnote:[https://amazon-ion.github.io/ion-docs/docs/spec.html#sexp]
+PartiQL should support s-expression types and values since PartiQL's type system is a superset over the Ion types. Because the current PartiQL specification does not formally define s-expressions operations, we consider the definition of collection index and wildcard steps on s-expressions as out-of-scope for this RFC.
```

Comment:
I would reword this section. I would avoid usage of terms like 'we'  and write it more formally and more tersely.

e.g.,

```
=== Limitations

* This RFC requires that every fully-qualified `<exclude path>` contain a root and at least one step.
* This RFC restricts tuple attribute exclude steps to use string literals and collection index exclude steps to use int literals. Thus `<exclude paths>` are statically known. This 
* This RFC makes no changes to schema and name inference, and assumes that such inference is run as a prerequisite.
* This RFC defines `<exclude paths>` only over `list`, `bag`, and `tuple` value collections.

```